duplicate_url_discarder_rules/queryRemoval/main.json

```diff
@@ -2419,8 +2448,8 @@
   },
   {
     "args": [
-      "productName",
-      "pid"
+      "pid",
+      "productName"
```

Comment:
First off, this is :+1:.

But also, looking at the website to verify that indeed the `pid` query parameter has no effect, I noticed that proper URL normalization (to detect duplicates before redirects) would need a new rule type to support treating `https://www.bunnings.com.au/ozito-1800w-2030psi-high-pressure-washer_p0254158` as `https://www.bunnings.com.au/_p0254158` (i.e. remove everything from the last URL path before the last `_`, at least when the last path ends in `_p<num>`). It’s not the first time I see something like that in an e-commerce website.

Reply:
Thanks for raising @Gallaecio. 

This normalization task seems to suit the `SubpathRemovalProcessor` but currently doesn't support substring removal in a given subpath. Created https://github.com/zytedata/duplicate-url-discarder/issues/24 for tracking.